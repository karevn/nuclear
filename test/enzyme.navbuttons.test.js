import React from 'react';
import { shallow } from 'enzyme';

import { NavButtons } from '../app/components/NavButtons';

describe('<NavButtons />', () => {
  let history;

  beforeEach(() => {
    history = {
      goBack: jest.fn(),
      goForward: jest.fn()
    };
  });

  it('Test back button', () => {
    history.index = 2;
    const wrapper = shallow(<NavButtons history={history} />);

    wrapper.at(0).childAt(0).simulate('click');
    expect(history.goBack).toHaveBeenCalled();
  });

  it('Test forward button', () => {
    history.index = 2;
    history.length = 4;
    const wrapper = shallow(<NavButtons history={history} />);

    wrapper.at(0).childAt(1).simulate('click');
    expect(history.goForward).toHaveBeenCalled();
  });

  it('Back button should be disabled when there is no previous page', () => {
    history.index = 1;
    const wrapper = shallow(<NavButtons history={history} />);

    expect(wrapper.at(0).childAt(0).hasClass('disable')).toBe(true);
  });

  it('Forward button should be disabled when there is no entry in history stack after current page', () => {
    history.index = 3;
    history.length = 4;
    const wrapper = shallow(<NavButtons history={history} />);

    expect(wrapper.at(0).childAt(1).hasClass('disable')).toBe(true);
  });
});
