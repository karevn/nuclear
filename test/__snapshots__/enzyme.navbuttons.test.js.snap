// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavButtons /> Back button should be disabled when there is no previous page 1`] = `
<div
  className="nav_buttons"
>
  <a
    className="disable"
    href="#"
  >
    <FontAwesome
      name="chevron-left"
    />
  </a>
  <a
    className="disable"
    href="#"
  >
    <FontAwesome
      name="chevron-right"
    />
  </a>
</div>
`;

exports[`<NavButtons /> Forward button should be disabled when there is no entry in history stack after current page 1`] = `
<div
  className="nav_buttons"
>
  <a
    className=""
    href="#"
    onClick={[MockFunction]}
  >
    <FontAwesome
      name="chevron-left"
    />
  </a>
  <a
    className="disable"
    href="#"
  >
    <FontAwesome
      name="chevron-right"
    />
  </a>
</div>
`;

exports[`<NavButtons /> Test back button 1`] = `
<div
  className="nav_buttons"
>
  <a
    className=""
    href="#"
    onClick={[MockFunction]}
  >
    <FontAwesome
      name="chevron-left"
    />
  </a>
  <a
    className="disable"
    href="#"
  >
    <FontAwesome
      name="chevron-right"
    />
  </a>
</div>
`;

exports[`<NavButtons /> Test forward button 1`] = `
<div
  className="nav_buttons"
>
  <a
    className=""
    href="#"
    onClick={[MockFunction]}
  >
    <FontAwesome
      name="chevron-left"
    />
  </a>
  <a
    className=""
    href="#"
    onClick={[MockFunction]}
  >
    <FontAwesome
      name="chevron-right"
    />
  </a>
</div>
`;
